src/common/hostPlatform.ts

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as child_process from "child_process";
+import {Node} from "./node/node";
+import * as path from "path";
+import * as Q from "q";
+
+/**
+ * Interface defining the host (desktop) platform specific operations.
+ */
+export interface IHostPlatform {
+    getUserHomePath(): string;
+    getSettingsHome(): string;
+    getCommand(packageName: string): string;
+    getExtensionPipePath(): string;
+    getPlatformId(): HostPlatformId;
+    killProcess(process: child_process.ChildProcess): Q.Promise<void>;
+    setEnvironmentVariable(name: string, value: string): Q.Promise<void>;
+}
+
+/**
+ * Defines the identifiers of all the platforms we support.
+ */
+export enum HostPlatformId {
+    WINDOWS,
+    OSX,
+    LINUX
+}
+
+/**
+ * IHostPlatform implemenation for the Windows platform.
+ */
+class WindowsHostPlatform implements IHostPlatform {
+    public getUserHomePath(): string {
+        return process.env.USERPROFILE;
+    }
+
+    public setEnvironmentVariable(name: string, value: string): Q.Promise<any> {
+        return Q.nfcall(child_process.exec, `setx ${name} ${value}`);
+    }
+
+    public getSettingsHome(): string {
+        return path.join(process.env.APPDATA, "vscode-react-native");
+    }
+
+    public getCommand(packageName: string): string {
+        return `${packageName}.cmd`;
```

Comment:
I would imagine that this is eventually going to break - at some point a tool we'll need to use will be an executable, for now it's probably ok. Some alternatives would be exposing a method for each command (e.g. getNpmCommand()) or taking an enum for the commands we understand (getCommand(COMMAND_NPM)).